project/source/shared/Spawner/FixedIntervalSpawner.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+#include "Spawner.h"
+
+namespace nc::sample
+{
+    /** A Spawner that operates on a regular interval. */
+    class FixedIntervalSpawner : public Spawner
```

Comment:
I don't know how to do file level comments in VSCode's github UI, but after looking through these I wonder if we shouldn't include them in the base engine.